Dev

nicolasmoreau opened the pull request.

nicolasmoreau committed:
rename 'next' buttons in guided query interface

nicolasmoreau committed:
consumer service choice is inside the result table

nicolasmoreau committed:
simplifies xsams processor choice in result table

nicolasmoreau committed:
removes refresh call to asyncform, as it is not active

nicolasmoreau committed:
- Improves portal presentation with CSS modifications
- Displays a presentation text on home page
- Displays 2 list of nodes : 1 for active nodes and 1 for inactive ones

nicolasmoreau committed:
add library

nicolasmoreau committed:
use an older version of sfl4j

nicolasmoreau committed:
add missing . at end of sentence

nicolasmoreau committed:
link to nodes on home page

nicolasmoreau committed:
improve position of elements in advanced query page, solves problems with firefox

nicolasmoreau committed:
changes tables width, removes useless css directive

nicolasmoreau committed:
enlarge the page for very long node names

nicolasmoreau committed:
add citation policy page

nicolasmoreau committed:
remove list of inactive nodes until the registry has been cleaned

nicolasmoreau committed:
reduces width of login form to match new page size

nicolasmoreau committed:
bug correction : problem when displaying saved results for a request with no response (default ivo value for EmptyResponse ivo://vamdc/none returned null value when asking for consumers)

nicolasmoreau committed:
type correction

nicolasmoreau committed:
Remove "Remove form" link from forms
Add a different view for all controllers instead of directly using standardForm (now it includes standardForm)

This is because of a problem of update in forms when clearing/removing :
Case 1
In advanced form : Add a Radiative form, add an Environment form
Then Clear function in Environment does not work

In advanced form : Add a Particle form, add an Atom form
Then Clear function in Atom does not work

If a form of the same type is added above another one, Clear does not work (because the ID of the new form is identical to the form below)

Case 2 :
In advanced form : Add 3 atom forms
Enter any value in Atom Symbol in each one (1, 2, 3)
Remove the first form
Then the values in the remaining forms are incorrect

To solve Case 1,  several views were added so that generated objects are differents in the page.
To solve Case 2, "Remove form" has been removed. A form can be hidden and a new function "Clear page" has been added in Advanced query

nicolasmoreau committed:
reduces and changes position of european logo

nicolasmoreau committed:
changes img size in footer

nicolasmoreau committed:
displays only Molecules in Molecule search form suggestion

nicolasmoreau committed:
corrections in sql requests

nicolasmoreau committed:
removes duplicated code

nicolasmoreau committed:
add base class for collisional forms

nicolasmoreau committed:
uses renamed methods

nicolasmoreau committed:
solves problem on Remove form functionality

nicolasmoreau committed:
Adds tests for QueryTreeController

nicolasmoreau committed:
correction in clear function call in formtitle form

nicolasmoreau committed:
adapt java entities to new species database

nicolasmoreau committed:
remove automatic opening of a popup when processor is called

nicolasmoreau committed:
New version of portal
 - connect to the new species database to find molecules
 - log requests to the nodes into the querystore
 - guided query "request type" menu option order has been changed (species in first)
 - buttons are now rectangular excepted in guided query menu

org.apache.httpcomponents has been added into portal.ejb pom.xml
a new Querytore class manage requests sent to the query store.
Querystore UL added in Settings class
A portal user agent has been added into Settings class

nicolasmoreau committed:
Remove trailing System.out

nicolasmoreau committed:
close BufferedReader in QueryStore class

nicolasmoreau committed:
remove comments

search only among molecules for loadMoleculesFromStoichForm and loadMoleculesFromName methods
in EntityFacade

nicolasmoreau committed:
- Do not open a blank tab when downloading a XSAMS file
- The string containing the request to the query store is built only once instead of being created in the query loop

nicolasmoreau committed:
Link to download xsams files replaced by a button, enabled only when query is done
Query store result is now displayed even with Firefox

Status of querystore query is displayed in the window

nicolasmoreau committed:
change entity annotation to point on prod database instead of dev

nicolasmoreau committed:
query querystore with https

nicolasmoreau committed:
update user agent value

nicolasmoreau committed:
parameter not renamed in api

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/entity/EntityFacade.java:
```diff
@@ -17,52 +17,56 @@
 				"INNER JOIN vs.vamdcSpeciesNameses vsn " +
-				"WHERE vsn.name = :Value";
+				"WHERE vsn.name = :Value " +
+				"AND vs.speciesType = 2 ";
```
Is it possible to avoid using magic numbers? Like, replace 2 by a named constant...

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/entity/EntityQuery.java:
```diff
@@ -21,7 +22,31 @@
+	
+	static Collection<String> suggestAtomName(EntityManager em,
+			String name) {
+		return suggestRestrictedSpeciesName(em, name, 1);	
```
same comment about magic numbers & constants

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/entity/species/VamdcInchikeyExceptions.java:
```diff
@@ -6,18 +6,22 @@
 import javax.persistence.Table;
+
 import org.hibernate.validator.NotNull;
+import org.vamdc.portal.Settings;
```
do you need this import here?

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/preview/PreviewManager.java:
```diff
@@ -51,12 +50,12 @@
 	public void initiate(){
 		
-		if (nodeFutureResponses.size()>0)
+		if (!nodeFutureResponses.isEmpty())
```
:+1:

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/queryBuilder/forms/MoleculesForm.java:
```diff
@@ -81,11 +80,7 @@ public void clear(){
-	public String info(){
-		return "test";
-	}
+	}	
```
oh, thank you for removing this

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/queryBuilder/forms/SpeciesForm.java:
```diff
@@ -1,13 +1,7 @@
-    	super();
-    }
-    
+public interface SpeciesForm extends Form{
```
:+1:

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/queryBuilder/formsTree/ReactionCollisionForm.java:
```diff
@@ -0,0 +1,163 @@
+		
+	}
+	
+	/*private void addHiddenForms(){
```
:scissors:

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/queryBuilder/formsTree/ReactionCollisionForm.java:
```diff
@@ -0,0 +1,163 @@
+	public void validate() {
+		//this.addHiddenForms();
+		
+		for(int i=0;i<typeAAtomCount;i++){
```
I'm wondering if in Java there is a fancier way to execute code N times, then writing a for loop? Probably not :(

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/queryBuilder/formsTree/SpeciesSelectionForm.java:
```diff
@@ -19,7 +19,7 @@
 	private QueryTreeInterface tree;
 	private Boolean queryable = true;
-	private Integer formsCount = 0;
+	//private Integer formsCount = 0;
```
:scissors:

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/queryLog/QueryStore.java:
```diff
@@ -0,0 +1,223 @@
+	/**
+	 * time between two request to the querystore, in milliseconds
+	 */
+	private Integer retryInterval = 3000;
```
Shouldn't we move it to Settings, instead of hardcoding the values?

misdoro commented on portal.ejb/src/main/java/org/vamdc/portal/session/queryLog/QueryStore.java:
```diff
@@ -0,0 +1,223 @@
+			
+		} else {
+			// empty response
+			if (response.getStatusLine().getStatusCode() == 204) {
```
wondering if `response.getStatusLine().getStatusCode()` should be extracted into a local variable

misdoro commented on portal.ejb/src/test/java/org/vamdc/portal/session/queryBuilder/forms/TestAddRemoveTreeForms.java:
```diff
@@ -0,0 +1,66 @@
+import org.vamdc.portal.session.queryBuilder.formsTree.RootForm;
+import org.vamdc.portal.session.queryBuilder.formsTree.SpeciesSelectionForm;
+
+public class TestAddRemoveTreeForms {
```
:+1:

misdoro reviewed (commented):
Thanks for the pull request.
I've left a few comments and blushed a couple of times when you corrected my old bad code.

nicolasmoreau committed:
improve code with Misha's remarks

nicolasmoreau committed:
fix problem :
	Display link appeared on all for all nodes when a processor had been selected for at least one and a request on the query store had been completed.
	Query store request completion now updates only the xsams cell on the node's line and not the whole page

nicolasmoreau committed:
improves concurrent execution of processors in multiple nodes

nicolasmoreau committed:
Fix problems caused by concurrent execution of multiple query store requests and/or xsams processors by
using Callable and Executor

nicolasmoreau committed:
add missinig package

nicolasmoreau committed:
improve error message display

nicolasmoreau committed:
improves error message display in result table

nicolasmoreau committed:
update display of saved requests page by removing stats column ( because result of requests can change over time)

nicolasmoreau committed:
Explicitely instanciate logger in class that are not Seam component

nicolasmoreau committed:
change text position on query save page

nicolasmoreau committed:
Merge branch 'dev'

nicolasmoreau committed:
remove comments

nicolasmoreau committed:
Species search selected by default in guided query mode

nicolasmoreau committed:
improve display of forms
fix bug on step by step interface : sometimes autosuggest would not work

nicolasmoreau committed:
When converting unit of a wavelength, result is converted to integer if double value is an integer.
This prevent display of trailing decimals like 10.000000001

nicolasmoreau committed:
Add unit test for WavelengthUnitConverter

nicolasmoreau committed:
Search by species is default option in guided query mode

nicolasmoreau committed:
Display "number of atoms  / molecules" form in columns instead of lines

nicolasmoreau committed:
Merge branch 'dev'

Conflicts:
	portal.ejb/src/main/java/org/vamdc/portal/session/queryLog/QueryStoreProcessor.java

nicolasmoreau committed:
When a request is sent to the querystore, its ip address is found in X-FORWARDED-FOR header.
However when going through a proxy, addresses of the client and the proxies are concatenated.

This correction keeps only the address of the original client if it is available

nicolasmoreau committed:
corrections of db name for dev version

nicolasmoreau committed:
Add a new Tools page and an entry in the main menu.

nicolasmoreau committed:
Modification in PreviewThread and PreviewManager classes to handle query to HTTPS nodes

nicolasmoreau committed:
Takes into account HEAD request returned with a status = 204 but values in HEADER.
A download lingk will be displayed in the result table

nicolasmoreau committed:
Add a privacy policy page

nicolasmoreau committed:
mark not recommended processor explicitely instead of using **

nicolasmoreau committed:
urlencode parameter value in request

nicolasmoreau committed:
https in request to node mirros

nicolasmoreau committed:
Modifications in portal to query https services, required a change in drool version.
This works with jdk1.8.0_152

nicolasmoreau committed:
removes duplicated artifact
